delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/eventlog/GlobalEventLog.scala

```diff
@@ -0,0 +1,34 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.eventlog
+
+import akka.persistence.query.Offset
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.ExpandedJsonLd
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRejection.ProjectNotFound
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Envelope, Event, ResourceF}
+import fs2.Stream
+import monix.bio.{IO, Task}
+
+trait GlobalEventLog {
+
+  /**
+    * Get stream of all [[Event]]s
+    * @param offset the offset to start from
+    */
+  def events(offset: Offset): Stream[Task, Envelope[Event]]
+
+  /**
+    * Get stream of all [[Event]]s for a project.
+    *
+    * @param project  the project reference
+    * @param offset   the offset to start from
+    */
+  def events(project: ProjectRef, offset: Offset): IO[ProjectNotFound, Stream[Task, Envelope[Event]]]
+
+  /**
+    * Fetch the latest for the event as expanded JSON-LD.
+    *
+    * @param event  the event for which to fetch latest state
+    */
+  def latestStateAsExpandedJsonLd(event: Event): Task[Option[ResourceF[ExpandedJsonLd]]]
```

Comment:
`latestState` probably is enough, or `expandedLatestState`

Reply:
Where do we need to use this method ?
In the tests, we use it with it after getting the stream of events ?
Maybe we could pass directly the envelope with the event which already contains the event type ?
This way we would not have to touch the Event contract ?

Reply:
We will need it in Blazegraph indexing to get the latest state for a given event. The eventType in `Envelope` is a specific event type(e.g. `ResourceCreated`), but here we need an event type which encompasses all event of a given type(all `ResourceEvent`s).